### PR TITLE
Use ServiceExport in agent

### PR DESCRIPTION
**Pull request by vthapar**
Use ServicExport to decide which services will be multicluster. This
changes the to an opt-in model from current all-in default.

### CODE REVIEW

File: test/e2e/dataplane/service_discovery.go [hunk 1]
```diff
@@ -4,34 +4,36 @@ import (
...
 	})
-
-	When("a pod tries to resolve a service which is present locally and in a remote cluster", func() {
-		It("should resolve the local service", func() {
-			RunServiceDiscoveryLocalTest(f)
+	/*
+		When("a pod tries to resolve a service which is present locally and in a remote cluster", func() {
+			It("should resolve the local service", func() {
+				RunServiceDiscoveryLocalTest(f)
+			})
```
Comment (skitt) [inline test/e2e/dataplane/service_discovery.go:31]: Why is this commented out?

Comment (vthapar) [reply test/e2e/dataplane/service_discovery.go:31]: Argh. Missed to uncomment after refactoring the dig method.

File: pkg/agent/main.go [hunk 2]
```diff
@@ -35,6 +36,10 @@ func main() {
 	if err != nil {
 		klog.Exitf("Error adding lighthouse V1 to the scheme: %v", err)
 	}
+	err = lighthousev2a1.AddToScheme(scheme.Scheme)
```
Comment (mangelajo) [inline pkg/agent/main.go:39]: This is something the generator does not do for some reason.

We need to register them in the register.go part, not here :)

https://github.com/submariner-io/submariner/blob/master/pkg/apis/submariner.io/v1/register.go#L28

@skitt also had a fun time about it.

Comment (vthapar) [reply pkg/agent/main.go:39]: > This is something the generator does not do for some reason.
> 
> We need to register them in the register.go part, not here :)
> 
> https://github.com/submariner-io/submariner/blob/master/pkg/apis/submariner.io/v1/register.go#L28
> 
> @skitt also had a fun time about it.

That is already done:
https://github.com/submariner-io/lighthouse/blob/master/pkg/apis/lighthouse.submariner.io/v2alpha1/register.go#L31

Without this line I was getting an error at runtime. We had to add it for multiclusterservices too.

Comment (mangelajo) [reply pkg/agent/main.go:39]: Hmm, may be it was somewhere else @skitt do you remember?

thanks vishal

**Pull request merged**